tcmba/spiders/obras.py

```diff
@@ -0,0 +1,114 @@
+from datetime import date, datetime
+
+from scrapy import Request, Spider
+
+from tcmba.items import ConstructionItem
+from tcmba.spiders.helpers import format_year
+
+
+class ConstructionsSpider(Spider):
+    name = "obras"
+    url = "https://www.tcm.ba.gov.br/portal-da-cidadania/obras/?municipio=2910800"
+    current_year = date.today().year
+
+    def __init__(self, *args, **kwargs):
+        super(ConstructionsSpider, self).__init__(*args, **kwargs)
+        if hasattr(self, "ano_inicial"):
+            try:
+                int(self.ano_inicial)
+                self.ano_inicial = int(format_year(self.ano_inicial))
+            except ValueError:
+                raise Exception("Você precisa informar um ano válido.")
+        else:
+            self.ano_inicial = date.today().year
+
+    def start_requests(self):
+        year = self.ano_inicial
+        while year <= self.current_year:
+            # 129 é o código de entidade da Prefeitura Mun. de Feira de Santana
+            url = f"{self.url}&ano={year}&entidade=129&pesquisar=Pesquisar"
+            year += 1
+            yield Request(url, self.parse)
+
+    def parse(self, response):
+        rows = response.css("table#tabelaResultado tbody tr")
+
+        for row in rows:
+            detail_url = (
+                row.css(".btn-acao::attr(onclick)")
+                .re_first(r"location.=\s*(.*)")
+                .replace("'", "")
+            )
+
+            company_content = row.css("a.btn-popover::attr(data-content)")
+
+            item = ConstructionItem(
+                construction_number=row.xpath(".//td[1]/text()").get(),
+                description=row.xpath(".//td[2]/text()").get(),
+                situation=row.xpath(".//td[3]/text()").get(),
+                type=row.xpath(".//td[4]/text()").get(),
+                work_or_service_value=row.xpath(".//td[5]/text()").get(),
+                paid_value=row.xpath(".//td[6]/text()").get(),
+                retained_value=row.xpath(".//td[7]/text()").get(),
+                executing_company={
+                    "name": company_content.re_first(r"<td>(\D+)</td>"),
```

Comment:
Aqui tu poderia pegar pegar a coluna usando o css path também:

```suggestion
                    "name": company_content.css("td"),
```